### ISSUE TITLE
Can't find Python executable "python"?

### ISSUE THREAD

**Issue by Gaubee** [Automattic/node-canvas#205]
I have downloaded and installed the Python (link address http://www.python.org/ftp/python/2.7.3/python-2.7.3.msi)
And I have to configure the environment variable, the variable named 'PYTHON', the variable value of the Python installation directory. I do not know why it has been suggested and can not find the environment variable. I have configuration problems or configuer.js of error?
—— Translate by Google

---------------------------------------------------
C:\Program Files\nodejs>npm install canvas
npm http GET https://registry.npmjs.org/canvas
npm http 304 https://registry.npmjs.org/canvas

> canvas@0.13.0 install C:\Program Files\nodejs\node_modules\canvas
> node-gyp rebuild

C:\Program Files\nodejs\node_modules\canvas>node "C:\Program Files\nodejs\node_m
odules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuil
d
gyp ERR! configure error
gyp ERR! stack Error: Can't find Python executable "python", you can set the PYT
HON env variable.
gyp ERR! stack     at failNoPython (C:\Program Files\nodejs\node_modules\npm\nod
e_modules\node-gyp\lib\configure.js:110:14)
gyp ERR! stack     at C:\Program Files\nodejs\node_modules\npm\node_modules\node
-gyp\lib\configure.js:74:11
gyp ERR! stack     at Object.oncomplete (fs.js:297:15)
gyp ERR! System Windows_NT 6.1.7601
gyp ERR! command "node" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modu
les\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Program Files\nodejs\node_modules\canvas
gyp ERR! node -v v0.8.6
gyp ERR! node-gyp -v v0.6.3
gyp ERR! not ok
npm ERR! canvas@0.13.0 install: `node-gyp rebuild`
npm ERR! `cmd "/c" "node-gyp rebuild"` failed with 1
npm ERR!
npm ERR! Failed at the canvas@0.13.0 install script.
npm ERR! This is most likely a problem with the canvas package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls canvas
npm ERR! There is likely additional logging output above.

npm ERR! System Windows_NT 6.1.7601
npm ERR! command "C:\\Program Files\\nodejs\\\\node.exe" "C:\\Program Files\\nod
ejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "canvas"
npm ERR! cwd C:\Program Files\nodejs
npm ERR! node -v v0.8.6
npm ERR! npm -v 1.1.48
npm ERR! code ELIFECYCLE
npm ERR!
npm ERR! Additional logging details can be found in:
npm ERR!     C:\Program Files\nodejs\npm-debug.log
npm ERR! not ok code 0
---------------------------------------------------------------

**Comment by duochan** [comment]
你是怎么解决的，我的环境变量也设置了，但只要npm就会报错~

**Comment by BoykaZhu** [comment]
直接敲  >set PYTHON=C:\Dev\Python27\python.exe

**Comment by hu1991die** [comment]
Does your problem have been resolved now?,I am also encountering the problem also..

**Comment by iaarnio** [comment]
TL;DR  Make a copy or alias of your python.exe with name python2.7.exe 

My python 2.7 was installed to directory D:\app\Python27. The executable was named python.exe.

I always got this error no matter how I set PYTHON env variable:

gyp ERR! stack Error: Can't find Python executable "python2.7", you can set the PYTHON env variable.
gyp ERR! stack     at failNoPython (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\configure.js:103:14)

The reason for this was that in node-gyp's configure.js the python executable was resolved like this:

var python = gyp.opts.python || process.env.PYTHON || 'python'

And it turned out that gyp.opts.python had value 'python2.7' thus overriding process.env.PYTHON.

I resolved this by creating an alias for python.exe executable with name node-gyp was looking for:

D:\app\Python27>mklink python2.7.exe python.exe

**Comment by shady333** [comment]
npm config set python d:\Python27\python.exe

this works for me

**Comment by LinusU** [comment]
Using `npm config set` to set the correct path to python should work...

e.g. `npm config set python d:\Python27\python.exe`

**Comment by SatishMHiremath** [comment]
I have faced and fixed this issue relates node-sass and python error.Try installing latest angular cli command with the steps below

Set the system variable path as well:
C:\Users\windowsusername\AppData\Roaming\npm

C:\Users\windowsusername\AppData\Roaming\npm\node_modules@angular\cli\bin;

C:\Program Files\nodejs;

Then start using angular cli command prompt using:
angular client project folder path>npm install -g @latest

angular client project folder path>npm install -g node-gyp
